native/acter/src/api/message.rs

```diff
@@ -82,11 +82,55 @@ use super::common::{
     AudioDesc, FileDesc, ImageDesc, LocationDesc, ReactionRecord, TextDesc, VideoDesc,
 };
 
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct EventSendState {
+    state: String,
+    error: Option<String>,
+    event_id: Option<OwnedEventId>,
+}
+
+impl EventSendState {
+    fn new(inner: &SdkEventSendState) -> Self {
+        let mut evt_id = None;
+        let mut err = None;
+        let state = match inner {
+            SdkEventSendState::NotSentYet => "NotSentYet".to_string(),
+            SdkEventSendState::Cancelled => "Cancelled".to_string(),
+            SdkEventSendState::SendingFailed { error } => {
+                err = Some(error.to_owned().to_string());
+                "SendingFailed".to_string()
+            }
+            SdkEventSendState::Sent { event_id } => {
+                evt_id = Some(event_id.clone());
+                "Sent".to_string()
+            }
+        };
+        EventSendState {
+            state,
+            error: err,
+            event_id: evt_id,
+        }
```

Comment:
minor style thing. I prefer using direct assignments rather than mutable variables. I'd propose:

```suggestion
        let (state, error, event_id) = match inner {
            SdkEventSendState::NotSentYet => ("NotSentYet".to_string(), None, None),
            SdkEventSendState::Cancelled => ("Cancelled".to_string(), None, None),
            SdkEventSendState::SendingFailed { error } =>
                ("SendingFailed".to_string(),  Some(error.to_owned().to_string()), None),
            SdkEventSendState::Sent { event_id } =>
                ("Sent".to_string(), None, Some(event_id.clone())),
        };
        EventSendState {
            state,
            error,
            event_id,
        }
```